src/services/PostDisplayService.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+class PostDisplayService
+{
+    static public function displayAllPosts(array $posts): string
+    {
+        if ($posts==[]){
+            $output = "<p class='text-2xl mb-2'>No posts found</p>";
```

Comment:
You should be able to just return this string here, which will elimate the need for the else.

So it becomes:

```
if ($posts == []) {
    return "...."
}

$output = ' ';

foreach.....
```